src/vs/workbench/api/node/extHost.protocol.ts

```diff
@@ -159,6 +159,9 @@ export abstract class MainThreadLanguageFeaturesShape {
 	$registerHoverProvider(handle: number, selector: vscode.DocumentSelector): TPromise<any> { throw ni(); }
 	$registerDocumentHighlightProvider(handle: number, selector: vscode.DocumentSelector): TPromise<any> { throw ni(); }
 	$registerReferenceSupport(handle: number, selector: vscode.DocumentSelector): TPromise<any> { throw ni(); }
+	// TODO(nick): Should this support sending more than one location at a time?
```

Comment:
Probably, I imagine the langserver might collect references by file in which case there would be multiple. I see no reason why not to pass an array.